src/gainmap.c

```diff
@@ -658,6 +655,21 @@ avifResult avifRGBImageComputeGainMap(const avifRGBImage * baseRgbImage,
         }
     }
 
+    // Compute the headroom for the base and alterate images. The gainmap is the log-ratio of the
+    // higher-headroom (more HDR) image to the lower-headroom (less HDR) image. Reverse the gainmap
+    // if this is the case.
+    gainMapMetadata.baseHdrHeadroom = log2f(AVIF_MAX(baseMax, kEpsilon));
+    gainMapMetadata.alternateHdrHeadroom = log2f(AVIF_MAX(altMax, kEpsilon));
+    if (gainMapMetadata.baseHdrHeadroom > gainMapMetadata.alternateHdrHeadroom) {
+        for (int j = 0; j < height; ++j) {
+            for (int i = 0; i < width; ++i) {
+                for (int c = 0; c < numGainMapChannels; ++c) {
```

Comment:
Nit: Since `c` is the index for the first dimension of the `gainMapF` array, it would be better to let `c` be the outer loop. See the nested for loop at line 697 for an example.